### ISSUE TITLE
Confuse with simulate mouse move and click

### ISSUE THREAD

**Issue by genius001** [scrapinghub/splash#433]
Some case, we have to simulate moues events move or click random bypass the analysis script. 
Because this analysis script record mouse parameters and send to server as a 1pix*1pix gif.

For example:  
```
http://example.com/1.gif?mouse-x=200&mouse-y=300&width=1024&height=768
```

According some previous posts, it was difficult to click an element using splash:runjs(snippet).
But splash:runjs/splash:evaljs/splash:jsfunc are script level. 

Is it possible develop some mouse feature on pyqt level?
e.g. splash:mouse_move(x, y) splash:mouse_click(x, y).

**Comment by kmike** [comment]
@genius001 yes, that'd be good to expose 'real' events to splash; implementation could use http://doc.qt.io/qt-5/qinputevent.html subclasses like QMouseEvent and QKeyEvent.

**Comment by pawelmhm** [comment]
I started to play around with this (as I also need this functionality recently) and done this: https://github.com/scrapinghub/splash/compare/master...pawelmhm:hover-click?expand=1 it adds unit test and sample implementation on browser_tab object. Selecting element is easy but seems like mouse click event is not fired properly when I do `_qtapp.postEvent(view, event)` perhaps I'm missing something there? Comments are welcome.

**Comment by Youwotma** [comment]
@pawelmhm phantomjs implements clicking as two separate events (mousedown and mouseup), maybe you could try that: https://github.com/ariya/phantomjs/blob/842715be9d2bb27865c179c12761290fa3f2929c/src/webpage.cpp#L1545 (also the rest of that function might give you some ideas of what's wrong)

**Comment by genius001** [comment]
Hi, @pawelmhm 

I'm Agree with @Youwotma 
```
QMouseEvent(QEvent.MouseButtonPress, pos, Qt.LeftButton, Qt.LeftButton, Qt.NoModifier)
QMouseEvent(QEvent.MouseButtonRelease, pos, Qt.LeftButton, Qt.LeftButton, Qt.NoModifier)
```
This code worked for me.

**Comment by kmike** [comment]
@genius001 https://github.com/scrapinghub/splash/pull/438 is merged, so I'm closing this ticket. `splash:mouse_hover`, `splash:mouse_click`, `splash:mouse_press` and `splash:mouse_release` methods will be available in Splash 2.1. Thanks!
